lib/apollo_tracing/middleware.ex

```diff
@@ -3,52 +3,38 @@ defmodule ApolloTracing.Middleware do
   Documentation for ApolloTracing.
   """
 
-  alias ApolloTracing.{Schema, Schema.Execution, Schema.Execution.Resolver}
+  alias ApolloTracing.Schema.Execution.Resolver
   alias Absinthe.Resolution
 
   # Called before resolving
-  def call(%Resolution{state: :unresolved} = res, _config) do
-    %{acc: %{
-      apollo_tracing_start_time: start_mono_time,
-      apollo_tracing: %Schema{
-        execution: %Execution{
-          resolvers: resolvers_so_far
-        }
-      }
-    }} = res
-
+  # if there isn't an `ApolloTracing` flag set then we aren't actually doing any tracing
+  def call(%Resolution{acc: %{apollo_tracing_start_time: start_mono_time}, state: :unresolved} = res, _config) do
     now = System.monotonic_time()
     resolver = %Resolver{
       path: Absinthe.Resolution.path(res),
       parentType: res.parent_type.name,
       fieldName: res.definition.name,
       returnType: Absinthe.Type.name(res.definition.schema_node.type, res.schema),
       startOffset: now - start_mono_time,
-      duration: nil
     }
 
-    res = put_in(
-      res.acc.apollo_tracing.execution.resolvers,
-      [resolver | resolvers_so_far]
-    )
-
-    %{res | middleware:
-       res.middleware ++ [{{__MODULE__, :after_field}, [start_time: now]}]
+    %{res |
+      extensions: Map.put(res.extensions, __MODULE__, resolver),
+      middleware: res.middleware ++ [{{__MODULE__, :after_field}, [start_time: now]}]
      }
   end
+  def call(res, _) do
+    res
+  end
 
   # Called after each resolution to calculate the duration
   def after_field(%Resolution{state: :resolved} = res, [start_time: start_time]) do
-    %{acc: %{
-      apollo_tracing: %Schema{
-        execution: %Execution{resolvers: [resolver | prev_resolvers]}
-      }
-    }} = res
+    %{extensions: %{__MODULE__ => resolver}} = res
```

Comment:
Now that the field has actually completed, I pull the resolver info for THIS specific field off of its extensions, and then on line 38 I can push that onto your `execution.resolvers` stack.